From e3ba09b35aa1e4a4bb454977e6f640ac7abe6d2d Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Fri, 24 Jul 2020 13:55:44 +0800
Subject: [PATCH 3/8] audio: scx15: Fix VBC firmware generation

The vbc_eq file at /data/local/media is autogenerated by the audio hal
everytime it starts and gets loaded by the hardware eventually.

In the stock rom, the /vendor/firmware/vbc_eq is actually a symlink to
/data/local/media and this regenerated everytime the audio hal is initialized.

In most of our custom roms, /vendor/firmware/vbc_eq is a plain ol file --
a precompiled one from the stock rom. Which should be fortunate to all of us.
Why?

Firmware generation in this source code is BROKEN. Loading this firmware will
effectively disable any sound, at least from the main speaker and headset.
Those using the NULL codec doesn't seem to be effected.

This patch erases the stub functions that should have been migrated to the
libvbeffects library. When not removed, these stub functions override
the full implementations at libvbeffects and prevents some
various parts of the firmware to be initialized
and configured properly.

Yep, even the function calls from libvbeffects gets stubbed.

Change-Id: I2781e507a28da4f98837606220cd6d9dfef64588
Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 audio/scx15/record_process/aud_proc_config.c | 32 +-------------------
 1 file changed, 1 insertion(+), 31 deletions(-)

diff --git a/audio/scx15/record_process/aud_proc_config.c b/audio/scx15/record_process/aud_proc_config.c
index e82f5bf..4a236d0 100644
--- a/audio/scx15/record_process/aud_proc_config.c
+++ b/audio/scx15/record_process/aud_proc_config.c
@@ -24,6 +24,7 @@
 
 #include "aud_proc.h"
 #include "aud_filter_calc.h"
+#include "vb_hal.h" // from libvbeffect
 
 
 /**---------------------------------------------------------------------------*
@@ -199,37 +200,6 @@ extern "c"
 
 
 
-/**---------------------------------------------------------------------------*
- **                     static Function definitions                             *
- **---------------------------------------------------------------------------*/
- void VB_SetHpfMode(BOOLEAN a)
-{
-}
-
-void VB_SetHpfParas(int16_t a0, int16_t a1, int16_t a2, int16_t a3, int16_t a4, int16_t a5, int16_t a6)
-{
-}	 
-
-void VB_SetHpfWidth(uint16_t width)
-{
-} 
-
-void  VB_SetHpfGain(int16_t gIndex, int16_t gValue)
-{
-}
-
-void VB_SwitchHpf(BOOLEAN is_enable)
-{
-}
-
-uint32_t VB_GetHpfGain(int16_t g_index)
-{
-    return 0;		
-}
-
-void VB_SetHpfLimit(int8_t limit)
-{
-}
 			
 /*****************************************************************************/
 //  Description:    set input gain of agc (iAgcIngainSet)
-- 
2.17.1

